### PR TITLE
Cartesi Machine SDK 0.17.1

### PR DESCRIPTION
**Pull request by endersonmaia**
Related to #300

### CODE REVIEW

**Comment by tuler** [comment]
`sunodo build` started to work.

Using templates from https://github.com/sunodo/sunodo-templates/pull/37

File: packages/sdk/Dockerfile [hunk 1]
```diff
@@ -34,26 +41,28 @@ FROM builder as su-exec
 
 # v0.2 -> f85e5bde1afef399021fbc2a99c837cf851ceafa
 WORKDIR /usr/local/src
-RUN <<EOF
-set -e
-curl -sL https://github.com/ncopa/su-exec/archive/f85e5bde1afef399021fbc2a99c837cf851ceafa.tar.gz | tar --strip-components=1 -zxvf -
-make
-EOF
+ADD --keep-git-dir=true https://github.com/ncopa/su-exec.git#f85e5bde1afef399021fbc2a99c837cf851ceafa /usr/local/src
```
Comment (tuler) [inline packages/sdk/Dockerfile:44]: any reason for the `--keep-git-dir`?

File: packages/sdk/Dockerfile [hunk 2]
```diff
@@ -67,18 +76,21 @@ ENV LANGUAGE en_US:en
 
 COPY entrypoint.sh /usr/local/bin/
 COPY retar /usr/local/bin/
-COPY --from=genext2fs /usr/local/bin/genext2fs /usr/bin/
+COPY --from=genext2fs /usr/local/bin/xgenext2fs /usr/bin/
```
Comment (tuler) [inline packages/sdk/Dockerfile:79]: why is this not going to `/usr/local/bin`?

**Review by tuler — COMMENTED** [review]
We should separate a PR for SDK 0.4. Which is very close to approval.
And publish it.

**Comment by tuler** [comment]
Separated CLI changes to PR https://github.com/sunodo/sunodo/pull/431

**Comment by tuler** [comment]
Should we add the rollups node in the SDK, and use the SDK in the CLI compose?
This can facilitate #193

File: packages/sdk/Dockerfile (hunk 1 above)
Comment (endersonmaia) [reply packages/sdk/Dockerfile:44]: In this su-exec case, no particular reason.

File: packages/sdk/Dockerfile (hunk 2 above)
Comment (endersonmaia) [reply packages/sdk/Dockerfile:78]: It will work anyway, but for better semantics and consistency, we could put at `/usr/local/bin`.

**Comment by endersonmaia** [comment]
> Should we add the rollups node in the SDK, and use the SDK in the CLI compose? This can facilitate #193

If we follow this path, should `sunodo/sdk` derive `FROM cartesi/rollups-node` instead?

**Comment by tuler** [comment]
> If we follow this path, should `sunodo/sdk` derive `FROM cartesi/rollups-node` instead?

If we follow this path, yes.

What are the pros and cons?
Supposing we have a flag in `sunodo run`, which can be hidden from regular users, that allows a custom validator image.

If SDK includes node:
- In case of a new emulator version we need to update both the SDK and CLI (to change the default version)
- In case of a new node we need to update both the SDK and CLI as well
- In case of a advanced user he needs to create a custom SDK image and use that in `sunodo run`

if SDK does not include node:
- In case of a new emulator version we need to update both the SDK and CLI (same thing)
- In case of a new node we need to update only the CLI (easier)
- In case of a advanced user he needs to create a custom SDK image and use that in `sunodo run` (same thing)

Right?

So no advantages I guess. Just need to allow sunodo run to use a custom image.

**Comment by endersonmaia** [comment]
> So no advantages I guess. Just need to allow sunodo run to use a custom image.

Is there already an issue to track that?

**Comment by tuler** [comment]
> Is there already an issue to track that?

Not specifically that, but we have #193

File: packages/sdk/Dockerfile (hunk 1 above)
Comment (tuler) [reply packages/sdk/Dockerfile:44]: Can I remove all `--keep-git-dir` from this Dockerfile?
This and the one from genext2fs

Comment (endersonmaia) [reply packages/sdk/Dockerfile:44]: What your concern here?

I think you could remove it. 

Some build processes use the git workspace to gather information about versioning and stuff, don't know if that's the case here.

Comment (tuler) [reply packages/sdk/Dockerfile:44]: I don't think that's the case for theses builds.
Just wanted to remove an explicit option if we don't explicitly need it.

File: packages/sdk/Dockerfile [hunk 3]
```diff
@@ -3,27 +3,34 @@ ARG BASE_IMAGE
...
+    ca-certificates \
+    curl \
+    libarchive-dev \
+    libtool
 rm -rf /var/lib/apt/lists/*
 EOF
 
 FROM builder as genext2fs
 
+# main -> a7abe9ecaeb376babc4148c1122dfb9103539d2c
```
Comment (endersonmaia) [inline packages/sdk/Dockerfile:29]: Let's use the tagged commit

`v1.5.3 -> 1c1992153eb77cc59a8447c631b141c955d37c2a `

Comment (tuler) [reply packages/sdk/Dockerfile:29]: The tagged commit does not have the latest fix @mpolitzer did

Comment (endersonmaia) [reply packages/sdk/Dockerfile:29]: maybe we should ask him to make a release

Comment (endersonmaia) [reply packages/sdk/Dockerfile:29]: but ok, if it doesn't have any other change, the tag will eventually point to the same commit

Comment (endersonmaia) [reply packages/sdk/Dockerfile:29]: anyway, the branch name on @mpolitzer's fork is `cartesi`.

Comment (tuler) [reply packages/sdk/Dockerfile:29]: I don't know if he has more commits to do to make it "1.5.4". Isn't there a `--version` that needs a change?

Comment (endersonmaia) [reply packages/sdk/Dockerfile:29]: probably https://github.com/cartesi/genext2fs/blob/cartesi/configure.ac#L5

Comment (tuler) [reply packages/sdk/Dockerfile:29]: https://github.com/cartesi/genext2fs/commit/1c1992153eb77cc59a8447c631b141c955d37c2a

**Review by tuler — APPROVED** [review]

**Pull request merged**